### PR TITLE
MM-37299 - invite members ab testing

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
This PR adds the invite members button call to action. The button will be shown in different places and with different shapes depending on the value set in the feature flag InviteMembersButton.
The options are:
sticky_button: sticked to the bottom of the left hand side panel
lhs_button: Just below of the direct messages users list
user_icon: at the right side of the plus (more about channels) icon at the top of the left hand side panel
none: don't show any button

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37299

#### Related Pull Requests
https://github.com/mattermost/mattermost-server/pull/18016

#### Screenshots
User Icon:
Captura de pantalla 2021-07-28 a las 21 18 30

Sticky:
Captura de pantalla 2021-07-28 a las 21 18 57

Lhs_button:
Captura de pantalla 2021-07-28 a las 21 19 09

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: components/sidebar/invite_members_button.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import {ModalIdentifiers} from 'utils/constants';
+
+import {getInviteMembersButtonLocation} from 'mattermost-redux/selectors/entities/preferences';
+
+type Props = {
+    buttonType: string;
+};
+
+const InviteMembersButton: React.FC<Props> = (props: Props): JSX.Element | null => {
```
Comment (marianunez) [inline components/sidebar/invite_members_button.tsx:25]: Let's add some unit tests for this new component

**Review by marianunez — CHANGES_REQUESTED** [review]
Looking great! 🎉 Would be ideal if we can also get a cypress test for these cases

**Comment by matthewbirtch** [comment]
@pablovelezvidal how can I get the different tests to display so I can review?

**Comment by pablovelezvidal** [comment]
> @pablovelezvidal how can I get the different tests to display so I can review?

Im working on getting the splitio values read from the spinwick server, once I get this done, I will let you know. If I can not get this done quickly, I can simulate the splitio value responses so you can validate the UX for this PR. I will keep you posted. @matthewbirtch

**Review by AGMETEOR — APPROVED** [review]
LGTM

**Review by nickmisasi — APPROVED** [review]

**Comment by pablovelezvidal** [comment]
@stevemudie @matthewbirtch This other PR https://github.com/mattermost/mattermost-webapp/pull/8506 contains this same code + a pipeline change so the spinwick server is updated correctly. You can do your tests there. Currently the default value for splitio is User Icon. Let me know via DM when you want me to change the values so you can see the different variations the button has.

**Review by stevemudie — APPROVED** [review]
Worked with Pablo to test the different invite button locations and everything worked as expected.

Thanks @pablovelezvidal !

**Comment by matthewbirtch** [comment]
Hey @pablovelezvidal a few things I've found:

### User Icon Option
1. The new icon next to the `+` icon is misaligned. It's sitting 1px too low
image

### Sticky Button Option
1. The button's border and text color should be full opacity of the sidebar text color.
2. The hover state will just be adding a background of 8% sidebar text color
2. The button size is too small. An update was made recently (sorry I don't think that got communicated back to you). It should be 32px in height.
3. The font-weight on the button label should be `font-weight: 600`
4. The font-size on the button label should be `font-size: 12px`
5. The font-size on the icon should be `font-size: 14px`
6. Can we add 4px of space between the icon and the label within the button?
7. Can we adjust the spacing around the button so that there is 16px of space all the way around it? And it looks like there is a min-width set on the button to 90%. If we make this padding adjustment, we should be able to make the button width 100% of its container

### LHS button version
1. Can we mimic the style of a channel list item for this button?  The icon size, the spacing should align with the channel items above
image

**Review by matthewbirtch — APPROVED** [review]
Nice work @pablovelezvidal. Based on the changes we reviewed together, I think we're good to go.

**Pull request merged**